doc/source/whatsnew/v1.5.0.rst

```diff
@@ -248,6 +248,7 @@ Conversion
 - Bug in :meth:`Series.astype` and :meth:`DataFrame.astype` from floating dtype to unsigned integer dtype failing to raise in the presence of negative values (:issue:`45151`)
 - Bug in :func:`array` with ``FloatingDtype`` and values containing float-castable strings incorrectly raising (:issue:`45424`)
 - Bug when comparing string and datetime64ns objects causing ``OverflowError`` exception. (:issue:`45506`)
+- Bug when attempting to replace ``numpy.nan`` values causing ``RecursionError``
```

Comment:
if i read the original issue correctly this then this was working in 1.4 so no actual change (its only a change in master), if so delete this note

Reply:
Yes you're right. I removed the note from the changelog.